### PR TITLE
refactor: add email notification

### PR DESCRIPTION
**Pull request by frankqianms**
need to set secrets later

### CODE REVIEW

File: .github/workflows/cli.yml [hunk 1]
```diff
@@ -50,3 +50,34 @@ jobs:
...
+          echo "to=frankqian@microsoft.com,chyuan@microsoft.com,shenwe@microsoft.com" >> $GITHUB_ENV
+          echo "body=Workflow URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}" >> $GITHUB_ENV
+          echo "subject=[APICENTER] Weekly APIC Extension Live Test Failed" >> $GITHUB_ENV
+
+      - name: Send Email Notification
+        run: |
+          response=$(curl \
+            --request POST \
+            --header "Content-Type: application/x-www-form-urlencoded" \
+            --data "grant_type=client_credentials&client_id=${{ secrets.MAIL_CLIENT_ID }}&client_secret=${{ secrets.MAIL_CLIENT_SECRET }}&resource=https://management.core.windows.net" \
```
Comment (blackchoey) [inline .github/workflows/cli.yml:72]: can we not have client secret here? Maybe use federated credential or other approach for authentication?

File: .github/workflows/cli.yml [hunk 2]
```diff
@@ -50,3 +50,34 @@ jobs:
...
+  notify:
+    if: failure()
+    runs-on: ubuntu-latest
+    needs: apic-extension-live-test
+    environment: engineering
+    steps:
+      - name: Generate Email Content
+        id: generate-email-content
+        run: |
+          echo "to=frankqian@microsoft.com,chyuan@microsoft.com,shenwe@microsoft.com" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (frankqianms) [reply .github/workflows/cli.yml:63]: sure

[the remark this answers is not in the document]
Comment (frankqianms) [reply .github/workflows/cli.yml:63]: @adashen shall we put emails to github secrets or envs in the yaml? which is better

File: .github/workflows/cli.yml (hunk 1 above)
Comment (blackchoey) [reply .github/workflows/cli.yml:72]: The security guideline is avoid using client secret, so let's try to follow the guideline if we can.

**Review by blackchoey — APPROVED** [review]
Please create follow up tasks for the comments

File: .github/workflows/cli.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (frankqianms) [reply .github/workflows/cli.yml]: put to github secrets first, since it is a public repo

**Pull request merged**